docs/book/src/Good Practices

```diff
@@ -0,0 +1,24 @@
+# Good Practices
+
+## What is "Reconciliation" in Operators?
+
+The reconciliation is a continuos  loop that performs necessary actions on the current state to ensure that the Custom Resource reaches the desired state as specified by the user.
+
+## Why should reconciliations be idempotent?
+When developing operators, the controller’s reconciliation loop needs to be idempotent. By following the [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) we create [controllers](https://kubernetes.io/docs/concepts/architecture/controller/) that provide  a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster. Developing idempotent solutions will allow the reconciler to correctly respond to generic or unexpected events, easily deal with application startup or upgrade. More explanation on this is available [here](https://github.com/kubernetes-sigs/controller-runtime/blob/main/FAQ.md#q-how-do-i-have-different-logic-in-my-reconciler-for-different-types-of-events-eg-create-update-delete). 
+
+Writing reconciliation logic according to specific events, breaks the recommendation of operator pattern and goes against the design principles of [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime). This may lead to unforeseen consequences, such as resources becoming stuck and requiring manual intervention.
+
+## Understanding Kubernetes APIs and following API conventions
+Building your operator commonly involves extending the Kubernetes API itself. It is helpful to understand precisely how Custom Resource Definitions (CRDs) interact with the Kubernetes API. Also, the [Kubebuilder documentation](https://book.kubebuilder.io/cronjob-tutorial/gvks.html) on Groups and Versions and Kinds may be helpful to understand these concepts better as they relate to operators.
```

Comment:
Could we please use the relative path for the link on docs? 
See how that works in the other ones. We should not use the link directly.